src/chai/ArrayManager.cpp

```diff
@@ -147,6 +155,12 @@ void ArrayManager::setExecutionSpace(ExecutionSpace space)
   CHAI_LOG(Debug, "Setting execution space to " << space);
   std::lock_guard<std::mutex> lock(m_mutex);
 
+#if defined(CHAI_ENABLE_PINNED)
+  if (chai::GPU == space) {
+    m_need_sync_for_pinned = true;
```

Comment:
How do you handle if the array is const on the device?

Reply:
The synchronize is only called when a move would normally occur for a non pinned MA, so all the same const semantics are preserved.